tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,157 @@
+// limitations under the License.
+
+package integrationoperator
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"maistra.io/istio-operator/pkg/util/tests/kubectl"
+	resourcecondition "maistra.io/istio-operator/pkg/util/tests/types"
+)
+
+var _ = Describe("Operator", Ordered, func() {
+	SetDefaultEventuallyTimeout(120 * time.Second)
+	SetDefaultEventuallyPollingInterval(time.Second)
+	var (
+		resourceAvailable = resourcecondition.Condition{
+			Type:   "Available",
+			Status: "True",
+		}
+		resourceReconcilied = resourcecondition.Condition{
+			Type:   "Reconciled",
+			Status: "True",
+		}
+
+		resourceReady = resourcecondition.Condition{
+			Type:   "Ready",
+			Status: "True",
+		}
+	)
+
+	Describe("operator installation", func() {
+		// TODO: we  need to support testing both types of deployment for the operator, helm and olm via subscription.
+		// Discuss with the team if we should add a flag to the test to enable the olm deployment and don't do that deployment in different step
+		When("default helm manifest are applied", func() {
+			BeforeAll(func() {
+				if skipDeploy == "true" {
+					Skip("Skipping the deployment of the operator and the tests")
+				}
+				GinkgoWriter.Println("Deploying Operator using default helm charts located in /chart folder")
+				Eventually(deployOperator).Should(Succeed(), "Operator deployment should be successful")
+			})
+
+			Specify("the operator is running", func() {
+				Eventually(kubectl.GetResourceCondition).WithArguments(namespace, "deployment", deploymentName).Should(ContainElement(resourceAvailable))
+				GinkgoWriter.Println("Operator deployment is Available")
+
+				Eventually(kubectl.GetPodPhase).WithArguments(namespace, "control-plane=istio-operator").Should(Equal("Running"), "Istio-operator pod should be Running")
+				GinkgoWriter.Println("Istio-operator pod is Running")
+			})
+		})
+	})
+
+	Describe("installation and unistallation of the istio resource", func() {
+		for _, version := range istioVersions {
+			// Note: This var version is needed to avoid the closure of the loop
+			version := version
+
+			Context(fmt.Sprintf("is applied the istio resource with version %s", version), func() {
+				BeforeAll(func() {
+					Expect(kubectl.CreateNamespace(controlPlaneNamespace)).To(Succeed())
+					Eventually(createIstioCR).WithArguments(version).Should(Succeed(), "Istio CR should be created")
+					GinkgoWriter.Println("Istio CR created")
+				})
+
+				AfterAll(func() {
+					GinkgoWriter.Println("Cleaning up")
+					Eventually(kubectl.DeleteNamespace).WithArguments(controlPlaneNamespace).Should(Succeed(), "Namespace should be deleted")
+					Eventually(kubectl.CheckNamespaceExist).WithArguments(controlPlaneNamespace).Should(MatchError(kubectl.ErrNotFound), "Namespace should be deleted")
+					GinkgoWriter.Println("Cleanup done")
+				})
+
+				When("the Istio resource is created", func() {
```

Comment:
Since this is within the context of `Describe("installation and uninstallation of the Istio resource`, we can shorted this to:
```suggestion
				When("the resource is created", func() {
```

Reply:
Same for all descriptions that are within this When (we don't need to repeat "Istio resource", because it's clear what "resource" refers to).